packages/app/providers/wallet-mobile-sdk-provider.tsx

```diff
@@ -0,0 +1,120 @@
+import { useCallback, useRef, useState, useEffect } from "react";
+
+import {
+  initiateHandshake,
+  makeRequest,
+  isConnected,
+  resetSession,
+} from "@coinbase/wallet-mobile-sdk";
+import { ethers } from "ethers";
+import { MMKV } from "react-native-mmkv";
+
+import { WalletMobileSDKContext } from "../context/wallet-mobile-sdk-context";
+
+const CB_WALLET_METADATA = {
+  name: "Coinbase Wallet",
+};
+
+const CACHED_ADDRESS_KEY = "mobile_sdk.address";
+const storage = new MMKV();
+
+type OnConnectedListener = (success: boolean) => void;
+
+interface WalletMobileSDKProviderProps {
+  children: React.ReactNode;
+}
+
+export function WalletMobileSDKProvider({
+  children,
+}: WalletMobileSDKProviderProps) {
+  const onConnectedListenerRef = useRef<OnConnectedListener | null>(null);
+  const [connected, setConnected] = useState<boolean>(false);
+  const [address, setAddress] = useState<string | null>(null);
+
+  const connect = useCallback(async () => {
+    try {
+      const action = {
+        method: "eth_requestAccounts",
+        params: {},
+      };
+      const [, account] = await initiateHandshake([action]);
+
+      const callback = onConnectedListenerRef.current;
+      if (account && account.address) {
+        setConnected(true);
+        setAddress(account.address);
+        storage.set(CACHED_ADDRESS_KEY, account.address);
+
+        callback && callback(true);
+        onConnectedListenerRef.current = null;
+      } else {
+        callback && callback(false);
+        onConnectedListenerRef.current = null;
+      }
+    } catch (error) {
+      resetSession();
+      setAddress(null);
+      setConnected(false);
+      onConnectedListenerRef.current = null;
```

Comment:
would we need to call `callback && callback(false);` here as well?

Reply:
Yep, good call. Addressed here [de1f532](https://github.com/showtime-xyz/showtime-frontend/pull/1463/commits/de1f53267ff1fc80fb392a39a32cc38067dc50ba)